quickannotator/db/crud/annotation.py

```diff
@@ -75,11 +72,15 @@ def __init__(self, image_id: int, annotation_class_id: int, is_gt: bool, in_work
         self.is_gt = is_gt
         self.scaling_factor = 1.0 if in_work_mag else base_to_work_scaling_factor(image_id, annotation_class_id)
 
-        if create_table:
-            model = self.create_annotation_table(image_id, annotation_class_id, is_gt)
-            self.model = model
+        table_name = build_annotation_table_name(image_id, annotation_class_id, is_gt=is_gt)
+        if table_exists(table_name):
+            self.model = create_dynamic_model(table_name)
         else:
-            self.model = create_dynamic_model(build_annotation_table_name(image_id, annotation_class_id, is_gt=is_gt))
+            if create_table:
```

Comment:
out of curiosity - why not create it when its accessed? its a lightweight operation, and i'm guessing 99.9% of the time people use the "get"  they would set create = True?

Reply:
code redundancy - this would require that the table checking & creation logic be moved to the top of each method.

I guess this logic could be performed in a function called `create_ifn_exists`, or even a decorator.

Thoughts?

Reply:
my question is more from an API user perspective - should they care/be aware of if the table is exists and/or should they have the requirement to dictate when the table is created? i.e., have we delegated too much unwarranted responsbility to them?

Reply:
I see what you mean. I guess we can just create the table if it doesn't exist. There's no need for the user to know the inner db interactions of AnnotationStore.

That said, I think we should add protections for the case where AnnotationStore is instantiated for a nonexistent annotation_class_id or image_id. In this case the __init__() function should raise an error.

Thoughts?

Reply:
yes, that makes sense to me  - are you happy with these conclusions then?